presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationFromAnnotationsParser.java

```diff
@@ -147,6 +150,25 @@ public static BindableAggregationFunction generateBindableAggregationFunction(Cl
                                 genericImplementations.build()));
     }
 
+    private static Optional<Method> getAggregationStateSerializerFactory(Class<?> aggregationDefinition, Class<?> stateClass)
+    {
+        // Only include methods that match this state class
+        List<Method> stateSerializerFactories = AnnotationHelpers.findPublicStaticMethodsWithAnnotation(aggregationDefinition, AggregationStateSerializerFactory.class).stream()
+                .filter(method -> ((AggregationStateSerializerFactory) method.getAnnotation(AggregationStateSerializerFactory.class)).value().equals(stateClass))
+                .collect(toImmutableList());
+
+        if (stateSerializerFactories.isEmpty()) {
+            return Optional.empty();
+        }
+
+        checkArgument(stateSerializerFactories.size() == 1,
+                String.format(
+                        "There must be up to one @AggregationStateSerializerFactory method in class %s for the @AggregationState %s ",
```

Comment:
`Expect at most 1 @AggregationStateSerializerFactory(%s.class) annotation, found %s in %s`

e.g. `Expect at most 1 @AggregationStateSerializerFactory(BlockState.class) annotation, found 2 in MaxAggregationFunction`